EZSwiftExtensionsTests/UITextFieldTests.swift

```diff
@@ -31,4 +32,118 @@ class UITextFieldTests: XCTestCase {
         
         XCTAssertEqual(textField.font?.pointSize, expectedFontSize)
     }
+    // MARK: Validations
+    func testValidateLength() {
+        let textField = UITextField(x: x, y: y, w: width, h: height, fontSize: 38)
+        var string = ""
+        
+        // empty
+        textField.text = string
+        XCTAssertFalse(textField.validateLength(ofCount: testLength, option: .equalTo))
+        XCTAssertFalse(textField.validateLength(ofCount: testLength, option: .greaterThan))
+        XCTAssertFalse(textField.validateLength(ofCount: testLength, option: .greaterThanOrEqualTo))
+        XCTAssertTrue(textField.validateLength(ofCount: testLength, option: .lessThan))
+        XCTAssertTrue(textField.validateLength(ofCount: testLength, option: .lessThanOrEqualTo))
+        
+        // == testLength
+        for _ in (0..<testLength) {
+            string.append("a")
+        }
+        textField.text = string // "aaaaa" (5 chars)
+        XCTAssertTrue(textField.validateLength(ofCount: testLength, option: .equalTo))
+        XCTAssertFalse(textField.validateLength(ofCount: testLength, option: .greaterThan))
+        XCTAssertTrue(textField.validateLength(ofCount: testLength, option: .greaterThanOrEqualTo))
+        XCTAssertFalse(textField.validateLength(ofCount: testLength, option: .lessThan))
+        XCTAssertTrue(textField.validateLength(ofCount: testLength, option: .lessThanOrEqualTo))
+        
+        // == nil; will be treated as ""
+        textField.text = nil
+        XCTAssertFalse(textField.validateLength(ofCount: testLength, option: .equalTo))
+        XCTAssertFalse(textField.validateLength(ofCount: testLength, option: .greaterThan))
+        XCTAssertFalse(textField.validateLength(ofCount: testLength, option: .greaterThanOrEqualTo))
+        XCTAssertTrue(textField.validateLength(ofCount: testLength, option: .lessThan))
+        XCTAssertTrue(textField.validateLength(ofCount: testLength, option: .lessThanOrEqualTo))
+    }
+    
+    func testValidateEmail() {
+        let textField = UITextField(x: x, y: y, w: width, h: height, fontSize: 38)
+        
+        // empty
+        textField.text = ""
+        XCTAssertFalse(textField.validateEmail())
+        
+        // valid format
+        textField.text = "lol@lol.lol"
+        XCTAssertTrue(textField.validateEmail())
+        for specialChar in "!#$%&'*+-/=?^_`{|}~".characters {
```

Comment:
Apparently you can have these special characters in the local part of an email address (the one before the `@`).

I was pretty surprise u can have that so this loop ensures each of them are tested at different scenarios (start of the email, end of the email, middle of the email).

I'm missing the scenario where there can be more than 1 of these special character in the same local part of an email. Ill add it in.